docs/RELEASING.md

```diff
@@ -0,0 +1,132 @@
+## Release process
+
+This section shows how to perform a release of Lokomotive. Only parts of the
+procedure are automated; this is somewhat intentional (manual steps for sanity
+checking) but it can probably be further scripted.
+
+We use goreleaser to automate some parts of the release, follow [the
+instructions](https://goreleaser.com/install/) to get it installed on your
+local machine.
+
+The following example assumes we’re going from version 0.1.0 (`v0.1.0`) to
+0.2.0 (`v0.2.0`). Check [Increasing version number](#increasing-version-number)
+for details on how to identify what the next version should be.
+
+- Start at the relevant milestone on GitHub (e.g.
+  https://github.com/kinvolk/lokomotive/milestones/v0.2.0): ensure all
+  referenced issues are closed or moved elsewhere. Close the milestone.
+
+- Create a release branch from latest master
+  - e.g. `git fetch origin && git checkout -b user/release-v0.2.0 origin/master`
+
+- Make sure your git status is clean.
+
+- Ensure the build is clean
+  - `git clean -ffdx && make all` should work.
+  - CI should be green.
+
+- Update the [release notes](CHANGELOG.md). Try to capture most of the salient
+  changes since the last release, but don't go into unnecessary detail (better
+  to link/reference the documentation wherever possible).
+  `scripts/changelog.sh` will help generating an initial list of changes.
+  Correct/fix entries if necessary, and group them by category.
+
+Even though it is set at build time, the Lokomotive version is also hardcoded
+in the repository, so the first thing to do is bump it:
+
+- Run `scripts/bump-release.sh v0.2.0`. This should generate two commits: a bump
+  to the actual release (e.g. v0.2.0, including CHANGELOG updates), and then a
+  bump to the release+git (e.g. v0.2.0+git). The actual release version should
+  only exist in a single commit! Sanity check what the script did with `git
+  diff HEAD^^` or similar.
+
+- If the script didn't work, yell at the author and/or fix it. It can almost certainly be improved.
+
+- File a PR and get a review from another maintainer. This is useful to a)
+  sanity check the diff, and b) be very explicit/public that a release is
+  happening.
+
+- Ensure the CI on the release PR is green!
+
+- Merge the PR.
+
+Check out the release commit
```

Comment:
```suggestion
Check out the release commit:
```

Reply:
Since this is like an "intermediate stop", I'll change it to:

```suggestion
Now we'll tag the release.
```

And write next:

```
- Check out the release commit, `git checkout HEAD^` should work. You want to
  be at the commit where the version is without "+git". Sanity check
  `pkg/version/version.go`.
```

Reply:
Cool, thanks!